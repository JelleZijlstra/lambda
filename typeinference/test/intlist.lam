data list a = Nil | Cons a (list a) in
let f = \l. match l with {
	Nil -> 0
	| Cons hd tl -> hd
} in
(f, (42 + (\x. x)))
