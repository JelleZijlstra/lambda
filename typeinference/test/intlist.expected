%%stdout%%
Error: Cannot unify types: 'typevar9 -> 'typevar9 and int
%%stderr%%
